spot_rl_experiments/spot_rl/envs/gaze_env.py

```diff
@@ -3,103 +3,194 @@
+
+    Args:
+        config (Config): Config object
+        dont_pick_up (bool): Whether to pick up the object or not
+        max_episode_steps (int): Max episode steps
+
+    Returns:
+        config (Config): Updated config object
+    """
+    # Update the config based on the input argument
+    config.DONT_PICK_UP = dont_pick_up
+
+    # Update max episode steps based on the input argument
+    config.MAX_EPISODE_STEPS = max_episode_steps
+    return config
 
-    env = SpotGazeEnv(config, spot)
-    env.power_robot()
-    policy = GazePolicy(config.WEIGHTS.GAZE, device=config.DEVICE)
-    policy.reset()
-    observations = env.reset(target_obj_id=target_obj_id)
-    done = False
-    env.say("Starting episode")
-    if orig_pos is None:
-        orig_pos = (float(env.x), float(env.y), np.pi)
-    while not done:
-        action = policy.act(observations)
-        observations, _, done, _ = env.step(arm_action=action)
-    # print("Returning to original position...")
-    # baseline_navigate(spot, orig_pos, limits=False)
-    # print("Returned.")
-    if done:
-        while True:
-            spot.set_base_velocity(0, 0, 0, 1.0)
-    return done
-
-
-def baseline_navigate(spot, waypoint, limits=True, **kwargs):
-    goal_x, goal_y, goal_heading = waypoint
-    if limits:
-        cmd_id = spot.set_base_position(
-            x_pos=goal_x,
-            y_pos=goal_y,
-            yaw=goal_heading,
-            end_time=100,
-            max_fwd_vel=0.5,
-            max_hor_vel=0.05,
-            max_ang_vel=np.deg2rad(30),
-        )
-    else:
-        cmd_id = spot.set_base_position(
-            x_pos=goal_x, y_pos=goal_y, yaw=goal_heading, end_time=100
-        )
-    cmd_status = None
-    success = False
-    traj = []
-    st = time.time()
-    while not success and time.time() < st + 20:
-        if cmd_status != 1:
-            traj.append((time.time(), *spot.get_xy_yaw()))
-            time.sleep(0.5)
-            feedback_resp = spot.get_cmd_feedback(cmd_id)
-            cmd_status = (
-                feedback_resp.feedback.synchronized_feedback.mobility_command_feedback
-            ).se2_trajectory_feedback.status
-        else:
-            if limits:
-                cmd_id = spot.set_base_position(
-                    x_pos=goal_x,
-                    y_pos=goal_y,
-                    yaw=goal_heading,
-                    end_time=100,
-                    max_fwd_vel=0.5,
-                    max_hor_vel=0.05,
-                    max_ang_vel=np.deg2rad(30),
-                )
-            else:
-                cmd_id = spot.set_base_position(
-                    x_pos=goal_x, y_pos=goal_y, yaw=goal_heading, end_time=100
-                )
 
-        x, y, yaw = spot.get_xy_yaw()
-        dist = np.linalg.norm(np.array([x, y]) - np.array([goal_x, goal_y]))
-        heading_diff = abs(wrap_heading(goal_heading - yaw))
-        success = dist < 0.3 and heading_diff < np.deg2rad(5)
+class GazeController:
```

Comment:
Thanks for designing this class! It looks nice. One design idea: right now, we are going to merge high-level skill PR, and do you think we can just call `GazeController` in `skill_manager` to init the policy and the `gaze_env`?

Reply:
Same thing for the place. I think you have place controller right?